### PR TITLE
doc: release-notes-latest: Add changes in drivers from upmerge 20210305

### PR DESCRIPTION
**Pull request by anangl**
Note changes related to Zephyr drivers that are relevant to NCS.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/releases/release-notes-latest.rst [hunk 1]
```diff
@@ -125,7 +125,51 @@ The current |NCS| release is based on Zephyr v2.5.99.
...
+  * Sensors:
+
+    Reworked the BME280 sensor driver to obtain device pointers directly (used :c:macro:`DEVICE_DT_GET` instead of :c:func:`device_get_binding`).
+    Made the QDEC nrfx driver usable on nRF5340 (added required devicetree and Kconfig entries).
+    Fixed an out of bounds write on the stack in the DPS310 sensor driver.
+    Added multi-instance support in the IIS2DLPC sensor driver.
+
+  * Serial:
+
+    Added waiting for the transmitter to go idle before powering down the UARTE peripheral in the asynchronous mode of the nRF UARTE driver.
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:163]:
```suggestion
    Updated the nRF UARTE driver to wait for the transmitter to go idle before powering down the UARTE peripheral in asynchronous mode.
```

File: doc/nrf/releases/release-notes-latest.rst [hunk 2]
```diff
@@ -125,7 +125,51 @@ The current |NCS| release is based on Zephyr v2.5.99.
...
+    Fixed the power down routine in the nRF UARTE driver. Now the RX interrupt is properly disabled.
+    Clarified the meaning of the ``timeout`` parameter of the :c:func:`uart_rx_enable` API function.
+
+  * USB:
+
+    Added Kconfig configuration of the stack size for the mass storage disk operations thread.
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:169]:
```suggestion
    Added Kconfig configuration of the stack size for the mass storage disk operations thread (:option:`CONFIG_MASS_STORAGE_STACK_SIZE`).
```

File: doc/nrf/releases/release-notes-latest.rst [hunk 3]
```diff
@@ -125,7 +125,51 @@ The current |NCS| release is based on Zephyr v2.5.99.
...
+    Added Kconfig configuration of inquiry parameters for the mass storage class.
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:170]:
```suggestion
    Added Kconfig configuration of inquiry parameters for the mass storage class (:option:`CONFIG_MASS_STORAGE_INQ_PRODUCT_ID`, :option:`CONFIG_MASS_STORAGE_INQ_REVISION`, :option:`CONFIG_MASS_STORAGE_INQ_VENDOR_ID`).
```

File: doc/nrf/releases/release-notes-latest.rst [hunk 4]
```diff
@@ -125,7 +125,51 @@ The current |NCS| release is based on Zephyr v2.5.99.
...
+    Fixed the initialization order in the ieee802154_nrf5 driver.
+    Corrected the pool from which RX packets are allocated in the ieee802154_nrf5 driver.
+    Added blocking on the RX packet allocation in the ieee802154_nrf5 driver to avoid dropping already acknowledged frames.
+    Add the :option:`CONFIG_IEEE802154_NRF5_UICR_EUI64_ENABLE` option to allow loading EUI64 from UICR registers.
+
+  * Sensors:
+
+    Reworked the BME280 sensor driver to obtain device pointers directly (used :c:macro:`DEVICE_DT_GET` instead of :c:func:`device_get_binding`).
+    Made the QDEC nrfx driver usable on nRF5340 (added required devicetree and Kconfig entries).
+    Fixed an out of bounds write on the stack in the DPS310 sensor driver.
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:158]:
```suggestion
    Fixed an out-of-bounds write on the stack in the DPS310 sensor driver.
```

**Review by ru-fu — CHANGES_REQUESTED** [review]

**Comment by anangl** [comment]
@ru-fu Suggested changes applied.

File: doc/nrf/releases/release-notes-latest.rst [hunk 5]
```diff
@@ -125,7 +125,51 @@ The current |NCS| release is based on Zephyr v2.5.99.
...
+    * Added a driver and a generic shield definition for Sharp memory displays of the LS0XX type.
+
+  * Flash:
+
+    * Implemented workaround for nRF52 anomaly 242 in the nRF SoC flash driver.
+    * Added automatic selection of :option:`CONFIG_MPU_ALLOW_FLASH_WRITE` when the MPU is enabled for Arm based SoCs.
+
+  * GPIO:
+
+    Used the nrfx GPIOTE channel allocator in the nRF GPIO driver to properly track GPIOTE channel allocations made in other modules.
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:144]:
```suggestion
    * Used the nrfx GPIOTE channel allocator in the nRF GPIO driver to properly track GPIOTE channel allocations made in other modules.
```

**Review by ru-fu — COMMENTED** [review]
One tiny issue left. ;)

File: doc/nrf/releases/release-notes-latest.rst (hunk 5 above)
Comment (anangl) [reply doc/nrf/releases/release-notes-latest.rst:144]: Right. How could I miss that one? I think I'm getting old. :wink:

Comment (anangl) [reply doc/nrf/releases/release-notes-latest.rst]: Fixed.

I see that one more change sneaked in (the order of the mass storage Kconfig options) with that update. I hope it won't be a problem.

**Review by ru-fu — APPROVED** [review]
Looks good, thanks! :)

**Comment by anangl** [comment]
Rebased to resolve a merge conflict.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**